learning_resources/models.py

```diff
@@ -0,0 +1,232 @@
+    """Platforms for all learning resources"""
+
+    platform = models.CharField(max_length=12, primary_key=True)
+    url = models.URLField(null=True, blank=True)
+    audience = models.CharField(
+        max_length=24,
+        choices=(
+            (constants.OPEN, constants.OPEN),
+            (constants.PROFESSIONAL, constants.PROFESSIONAL),
+        ),
+    )
+    is_edx = models.BooleanField(default=False)
+    has_content_files = models.BooleanField(default=False)
+
+    def __str__(self):
+        return self.platform
+
+
+class LearningResourceTopic(TimestampedModel):
+    """
+    Topics for all learning resources (e.g. "History")
+    """
+
+    name = models.CharField(max_length=128, unique=True)
+
+    def __str__(self):
+        return self.name
+
+
+class LearningResourceOfferor(TimestampedModel):
+    """Represents who is offering a learning resource"""
+
+    name = models.CharField(max_length=256, unique=True)
+
+    def __str__(self):
+        return self.name
+
+
+class LearningResourceImage(TimestampedModel):
+    """Represent image metadata for a learning resource"""
+
+    url = models.TextField(max_length=2048, null=True, blank=True)
+    description = models.CharField(max_length=1024, null=True, blank=True)
+    alt = models.CharField(max_length=1024, null=True, blank=True)
+
+    def __str__(self):
+        return self.url
+
+
+class LearningResourceDepartment(TimestampedModel):
+    """Department data for a learning resource"""
+
+    department_id = models.CharField(max_length=6, unique=True)
+    name = models.CharField(max_length=256, null=False, blank=False)
+
+    def __str__(self):
+        return self.name
+
+
+class LearningResourceContentTag(TimestampedModel):
+    """Represents course feature tags and contentfile types"""
+
+    name = models.CharField(max_length=128, unique=True)
+
+    def __str__(self):
+        return self.name
+
+
+class LearningResourcePrice(TimestampedModel):
+    """
+    Price model for all learning resources (e.g. "price": 0.00, "mode": "audit")
+    """
+
+    price = models.DecimalField(decimal_places=2, max_digits=12)
+    mode = models.CharField(max_length=128)
+    upgrade_deadline = models.DateTimeField(null=True, blank=True)
+
+    def __str__(self):
+        return "${:,.2f}".format(self.price)
+
+
+class LearningResourceInstructor(TimestampedModel):
+    """
+    Instructors for learning resources
+    """
+
+    first_name = models.CharField(max_length=128, null=True, blank=True)
+    last_name = models.CharField(max_length=128, null=True, blank=True)
+    full_name = models.CharField(max_length=256, null=True, blank=True, unique=True)
+
+    class Meta:
+        ordering = ["last_name"]
+
+    def __str__(self):
+        return self.full_name or " ".join((self.first_name, self.last_name))
+
+
+class LearningResource(TimestampedModel):
+    """Core model for all learning resources"""
+
+    object_id = models.CharField(max_length=128, null=False, blank=False)
```

Comment:
Sometimes it will be a readable id, depends on the source (some sources like prolearn don't really have a "readable" id, and for those sources this field is just used as a way to avoid dupes).  But I'd say it's a readable id more often than not, so I will change it.